lib/ansible/modules/database/mongodb/mongodb_user.py

```diff
@@ -213,20 +213,30 @@ def check_compatibility(module, client):
         module: Ansible module.
         client (cursor): Mongodb cursor on admin database.
     """
-    loose_srv_version = LooseVersion(client.server_info()['version'])
-    loose_driver_version = LooseVersion(PyMongoVersion)
+    COMPATIBILITY_TABLE = {
+        "4.2": "3.9",
+        "4.0": "3.7",
+        "3.6": "3.6",
+        "3.4": "3.4",
+        "3.2": "3.2",
+        "3.0": "2.8",
+        "2.6": "2.7",
+        "2.4": "2.5"
+    }
+
+    # NOTE: fishy way to retrieve major version
+    # e.g. 3.4.16 -> 3.4
+    srv_version = client.server_info()['version']
+    loose_srv_version = LooseVersion(''.join(srv_version.split())[:-3])
```

Comment:
I'm not sure what the purpose of the split and join currently are. Shouldn't this be splitting and joining on `.`?

```suggestion
    loose_srv_version = LooseVersion('.'.join(srv_version.split('.')[:2]))
```

Reply:
imho it's pretty similar as result, just the logical part is little bit different.

Reply:
It's different because the join and split actually do something and have a clear reason for being included. It also correctly handles versions like 3.4.1, which chopping off the end of the string doesn't.

```
>>> srv_version = '3.4.16'
>>> srv_version[:-3]
'3.4'
>>> srv_version = '3.4.1'
>>> srv_version[:-3]
'3.'
```